.eslintrc.js

```diff
@@ -227,12 +232,53 @@ module.exports = {
     },
     {
       files: ['*.tsx'],
+      excludedFiles: '*.spec.tsx',
+      rules: {
+        // WARNING: If updated, make sure these rules are merged with `no-restricted-imports` (#ts-source-files)
+        'no-restricted-imports': [
+          'error',
+          {
+            patterns: [
+              // Allow deeper imports for TypeScript types. TODO?
+              '@material-ui/*/*/*/*',
+              // Macros are fine since they're transpiled into something else
+              '!@material-ui/utils/macros/*.macro',
+            ],
+          },
+        ],
+        'react/prop-types': 'off',
+      },
+    },
+    // Files used for generating TypeScript declaration files (#ts-source-files)
+    {
+      files: ['**/packages/**/src/**/*.tsx'],
```

Comment:
Would this be enough?

```suggestion
      files: ['packages/*/src/**/*.tsx'],
```

Reply:
That makes more sense. Let me verify this locally real quick.

Reply:
I have used this in https://github.com/mui-org/material-ui/pull/24147/files#diff-e2954b558f2aa82baff0e30964490d12942e0e251c1aa56c3294de6ec67b7cf5R287

Reply:
Works. Same amount of problems reported

Reply:
Patch was corrupted